### PR TITLE
Add a consistent ARM64 image build process

### PR DESCRIPTION
**Pull request by rdimitrov**
## Description

Hello, 
The goal of the following changes is to update the ARM64 build support for part of the OpenFaaS services. 

The following projects are affected by that change (ticked ones are already merged):
- [ ] `faas` - this one
- [x] `faas-cli` - https://github.com/openfaas/faas-cli/pull/568
- [ ] `faas-idler` - https://github.com/openfaas-incubator/faas-idler/pull/24
- [x] `faas-netes` - https://github.com/openfaas/faas-netes/pull/331
- [x] `faas-swarm` - https://github.com/openfaas/faas-swarm/pull/40
- [ ] `nats-queue-worker` - https://github.com/openfaas/nats-queue-worker/pull/47
- [x] `openfaas-operator` - https://github.com/openfaas-incubator/openfaas-operator/pull/66
- [ ] `licence-check` - https://github.com/alexellis/license-check/pull/3

* The changes are mostly related to the Makefiles and the ARM64 specific Dockerfiles of each project.
* The make options follow the approach set by the ARMHF build script. 
* The script itself is also updated to support both ARMHF and ARM64 builds.

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?

* Triggered the `publish-arm.sh` script and every image passed successfully. 
* Then deployed OpenFaaS w/ Docker Swarm using these images.
* Triggering the `sha512sum` function went successful.  
* Triggering the `nodeinfo` function went with Bad Gateway. Though checking the service logs shows that the function is triggered and it works as expected. Didn't debug it further.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by alexellis** [comment]
@rdimitrov please can you make your issue into a check-list and show which PRs are merged?

File: contrib/publish-arm.sh [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 declare -a repos=("openfaas-incubator/openfaas-operator" "openfaas-incubator/faas-idler" "openfaas/faas" "openfaas/faas-swarm" "openfaas/nats-queue-worker" "openfaas/faas-netes" "openfaas/faas-cli")
 HERE=`pwd`
+ARCH=$(uname -m)
```
Comment (alexellis) [inline contrib/publish-arm.sh]: Was this tested on both variants?

File: deploy_stack.arm64.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
```
Comment (alexellis) [inline deploy_stack.arm64.sh]: Can you see if you can update deploy_stack.sh for @rgee0's latest change and incorporate that too? Perhaps we can avoid having to add this file.

File: docker-compose.arm64.yml [hunk 3]
```diff
@@ -1,39 +1,178 @@
...
+                limits:
+                    memory: 125M
+                reservations:
+                    memory: 50M
+            placement:
+                constraints:
+                    - 'node.platform.os == linux'
+
+    queue-worker:
+        image: openfaas/queue-worker:0.5.4-arm64
```
Comment (alexellis) [inline docker-compose.arm64.yml]: I got the following error:

```
docker pull openfaas/queue-worker:0.5.4-arm64
Error response from daemon: manifest for openfaas/queue-worker:0.5.4-arm64 not found
```

Should we merge or wait until I can publish the image?

File: docker-compose.arm64.yml [hunk 4]
```diff
@@ -1,39 +1,178 @@
...
+                constraints:
+                    - 'node.platform.os == linux'
+    # End services
+
+    # Start monitoring
+
     prometheus:
-        image: alexellis2/prometheus:1.5.2-arm64
-        command: "-config.file=/etc/prometheus/prometheus.yml -storage.local.path=/prometheus -storage.local.memory-chunks=10000 --alertmanager.url=http://alertmanager:9093"
+        image: alexellis2/prometheus:2.3.1-arm64
```
Comment (alexellis) [inline docker-compose.arm64.yml]: Docker image is valid and exists

File: docker-compose.arm64.yml [hunk 5]
```diff
@@ -1,39 +1,178 @@
...
+                delay: 5s
+                max_attempts: 20
+                window: 380s
+            placement:
+                constraints:
+                    - 'node.platform.os == linux'
+
+    # Docker Swarm provider
+    faas-swarm:
+        image:  openfaas/faas-swarm:0.5.0-arm64
```
Comment (alexellis) [inline docker-compose.arm64.yml]: Docker image is valid and exists

File: docker-compose.arm64.yml [hunk 6]
```diff
@@ -1,39 +1,178 @@
-version: "3.2"
+version: "3.3"
 services:
     gateway:
-        volumes:
-            - "/var/run/docker.sock:/var/run/docker.sock"
         ports:
             - 8080:8080
-        image: functions/gateway:0.7.0-arm64
+        image: openfaas/gateway:0.9.11-arm64
```
Comment (alexellis) [inline docker-compose.arm64.yml]: Docker image is valid and exists

File: docker-compose.arm64.yml [hunk 7]
```diff
@@ -1,39 +1,178 @@
...
+                    - 'node.role == manager'
+                    - 'node.platform.os == linux'
+            resources:
+                limits:
+                    memory: 500M
+                reservations:
+                    memory: 200M
+
+    alertmanager:
+        image: alexellis2/alertmanager-armhf:0.5.1
```
Comment (alexellis) [inline docker-compose.arm64.yml]: I think this is wrong.

**Comment by alexellis** [comment]
Derek add label: blocked

File: contrib/publish-arm.sh (hunk 1 above)
Comment (rdimitrov) [reply contrib/publish-arm.sh]: Yes, I believe. I'll give it a go these days once again though just to be sure 👍

**Comment by alexellis** [comment]
Thank you 🎉 !

**Pull request merged**